Doc/c-api/long.rst

```diff
@@ -529,10 +529,190 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
    Exactly what values are considered compact is an implementation detail
    and is subject to change.
 
+   .. versionadded:: 3.12
+
+
 .. c:function:: Py_ssize_t PyUnstable_Long_CompactValue(const PyLongObject* op)
 
    If *op* is compact, as determined by :c:func:`PyUnstable_Long_IsCompact`,
    return its value.
 
    Otherwise, the return value is undefined.
 
+   .. versionadded:: 3.12
+
+
+Import/Export API
+^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 3.14
+
+.. c:type:: Py_digit
+
+   A single unsigned digit in the range [``0``; ``PyLong_BASE - 1``].
+
+   It is usually used in an *array of digits*, such as the
+   :c:member:`PyLong_DigitArray.digits` array.
+
+   Its size depend on the :c:macro:`!PYLONG_BITS_IN_DIGIT` macro:
+   see the ``configure`` :option:`--enable-big-digits` option.
+
+   See :c:member:`PyLong_LAYOUT.bits_per_digit` for the number of bits per
+   digit and :c:member:`PyLong_LAYOUT.digit_size` for the size of a digit (in
+   bytes).
+
+
+.. c:struct:: PyLong_LAYOUT
+
+   Layout of an array of digits, used by Python :class:`int` object.
+
+   See also :attr:`sys.int_info` which exposes similar information to Python.
+
+   .. c:member:: uint8_t bits_per_digit
+
+      Bits per digit.
+
+   .. c:member:: uint8_t digit_size
+
+      Digit size in bytes.
+
+   .. c:member:: int8_t word_endian
+
+      Word endian:
+
+      - ``1`` for most significant byte first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+   .. c:member:: int8_t array_endian
+
+      Array endian:
+
+      - ``1`` for most significant byte first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+
+.. c:function:: PyObject* PyLong_Import(int negative, size_t ndigits, Py_digit *digits)
+
+   Create a Python :class:`int` object from an array of digits.
+
+   On success, return a Python :class:`int` object.
+   On error, set an exception and return ``NULL``.
+
+   *negative* is ``1`` if the number is negative, or ``0`` otherwise.
+
+   *ndigits* is the number of digits in the *digits* array.
+
+   *digits* is an array of unsigned digits. Digits must be in the range
+   [``0``; ``PyLong_BASE - 1``].
```

Comment:
This macro is undocumented.

BTW, commentary in longintrepr.h says: "Type 'digit' should be able to hold 2*PyLong_BASE-1":
```pycon
>>> _testcapi.pylong_export(2*(1<<sys.int_info.bits_per_digit)-1)
(0, [1073741823, 1])
>>> _testcapi.pylong_export(2*(1<<sys.int_info.bits_per_digit))
(0, [0, 2])
>>> _testcapi.pylong_import(0, [0, 2])
2147483648
>>> 2*(1<<sys.int_info.bits_per_digit)
2147483648
```
```suggestion
   *digits* is an array of unsigned digits. Digits must be in the range
   [``0``; ``2**(PyLong_SHIFT+1) - 1``], where :c:macro:`!PyLong_SHIFT`
   is equal to :c:member:`PyLong_LAYOUT.bits_per_digit`.
```

Reply:
> BTW, commentary in longintrepr.h says: "Type 'digit' should be able to hold 2*PyLong_BASE-1"

The `digit` type is used for computation, it should support a range larger than Python int digit range. Python int digits must be in the range [0; PyLong_BASE-1].